core/src/main/java/com/linecorp/armeria/client/retry/RetryRuleDecision.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import javax.annotation.Nullable;
+
+/**
+ * A {@link RetryRuleDecision} that determines whether a {@link RetryRule} is applied or not.
+ */
+public final class RetryRuleDecision {
+
+    private static final RetryRuleDecision STOP = new RetryRuleDecision(null);
+    private static final RetryRuleDecision NEXT = new RetryRuleDecision(null);
+
+    /**
+     * Returns a {@link RetryRuleDecision} that retries with the specified {@link Backoff}.
+     */
+    public static RetryRuleDecision retry(Backoff backoff) {
+        return new RetryRuleDecision(backoff);
+    }
+
+    /**
+     * Returns a {@link RetryRuleDecision} that never retries.
+     */
+    public static RetryRuleDecision stop() {
```

Comment:
Instead of `stop`, `noRetry`?

Reply:
`noRetry` seems more explicit and `stop` keeps consistency because other decision factory methods consist of a single word. 

Anyway, I prefer an obvious method name.